demo_nodes_py/demo_nodes_py/parameters/set_parameters_callback.py

```diff
@@ -0,0 +1,93 @@
+# Copyright 2022 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import rclpy
+from rclpy.executors import ExternalShutdownException
+from rclpy.node import Node
+from rclpy.parameter import Parameter
+
+from rcl_interfaces.msg import SetParametersResult
+
+
+# Example usage: changing param1 successfully will result in setting of param2.
+# ros2 service call /set_parameters_callback/set_parameters rcl_interfaces/srv/SetParameters
+#       "{parameters: [{name: "param1", value: {type: 3, double_value: 1.0}}]}"
+
+# node for demonstrating correct usage of pre_set, on_set
+# and post_set parameter callbacks
+class SetParametersCallback(Node):
+    
+    def __init__(self):
+        super().__init__('set_parameters_callback')
+        self.declare_parameter('param1', 0.0)
+        self.declare_parameter('param2', 0.0)
+
+        # tracks 'param1' value
+        self.internal_tracked_param_1 = self.get_parameter('param1').value
+
+        # tracks 'param2' value
+        self.internal_tracked_param_2 = self.get_parameter('param2').value
```

Comment:
I'm less certain in Python, but I do think that `self.declare_parameter` returns the `Parameter` object.  In that case, I think you can do:

```suggestion
        # tracks 'param1' value
        self.internal_tracked_param_1 = self.declare_parameter('param1', 0.0).value
        # tracks 'param2' value
        self.internal_tracked_param_2 = self.declare_parameter('param2', 0.0).value
```